src/c++/perf_analyzer/model_parser.cc

```diff
@@ -305,16 +288,36 @@ ModelParser::InitTorchServe(
 }
 
 cb::Error
-ModelParser::GetEnsembleSchedulerType(
-    const rapidjson::Document& config, const std::string& model_version,
-    std::unique_ptr<cb::ClientBackend>& backend, bool* is_sequential)
+ModelParser::DetermineComposingModelMap(
+    const std::vector<cb::ModelIdentifier>& bls_composing_models,
+    const rapidjson::Document& config,
+    std::unique_ptr<cb::ClientBackend>& backend)
+{
+  RETURN_IF_ERROR(AddBLSComposingModels(bls_composing_models, config));
+  RETURN_IF_ERROR(AddEnsembleComposingModels(config, backend));
+
+  return cb::Error::Success;
+}
+
+cb::Error
+ModelParser::AddBLSComposingModels(
+    const std::vector<cb::ModelIdentifier>& bls_composing_models,
+    const rapidjson::Document& config)
 {
-  const auto& sequence_itr = config.FindMember("sequence_batching");
-  if (sequence_itr != config.MemberEnd()) {
-    *is_sequential = true;
+  for (auto model : bls_composing_models) {
+    (*composing_models_map_)[config["name"].GetString()].insert(model);
```

Comment:
What if the bls composing model is an ensemble? We probably need to call `AddEnsembleComposingModels` in that case.

Reply:
Good catch. Fixed